### PR TITLE
Use pathlib for dir/file operations

### PR DESCRIPTION
**Pull request by eumiro**
Since the project is Python3 only, we can use `pathlib.Path` instead of all `os.path/glob/fnmatch/open` calls.

### CODE REVIEW

File: faker/sphinx/documentor.py [hunk 1]
```diff
@@ -104,8 +103,7 @@ def _write_base_provider_docs():
 
 def _write_standard_provider_docs():
     for provider_name in STANDARD_PROVIDER_NAMES:
-        fname = os.path.join(DOCS_ROOT, 'providers', '%s.rst' % provider_name)
-        with open(fname, 'wb') as fh:
+        with (DOCS_ROOT / 'providers' / '{}.rst'.format(provider_name)) as fh:
```
Comment (fcurella) [inline faker/sphinx/documentor.py:106]: We could use an f-string here:

```suggestion
        with (DOCS_ROOT / 'providers' / f'{provider_name}.rst') as fh:
```

File: faker/sphinx/documentor.py [hunk 2]
```diff
@@ -131,8 +128,7 @@ def _write_localized_provider_index():
 def _write_localized_provider_docs():
     for locale in AVAILABLE_LOCALES:
         info = _get_localized_provider_info(locale)
-        fname = os.path.join(DOCS_ROOT, 'locales', '{}.rst'.format(locale))
-        with open(fname, 'wb') as fh:
+        with (DOCS_ROOT / 'locales' / '{}.rst'.format(locale)).open('wb') as fh:
```
Comment (fcurella) [inline faker/sphinx/documentor.py:131]: we could use an f-string here too:
```suggestion
        with (DOCS_ROOT / 'locales' / f'{locale}.rst').open('wb') as fh:
```

**Review by fcurella — CHANGES_REQUESTED** [review]

File: faker/sphinx/documentor.py (hunk 1 above)
Comment (eumiro) [reply faker/sphinx/documentor.py:106]: There are plenty of f-strings there and I waited for the Python 3.6+ merge. I'll do all f-strings in the next step.

File: faker/sphinx/documentor.py (hunk 2 above)
Comment (eumiro) [reply faker/sphinx/documentor.py:131]: ditto

**Review by fcurella — APPROVED** [review]

**Comment by fcurella** [comment]
Thank you! ✨

**Pull request merged**